Modules/Core/Common/src/itkImageIORegion.cxx

```diff
@@ -17,45 +17,42 @@
  *=========================================================================*/
 
 #include "itkImageIORegion.h"
+#include <algorithm> // For copy_n.
 
 namespace itk
 {
-ImageIORegion ::ImageIORegion()
-{
-  m_ImageDimension = 2;
-  m_Index.resize(2);
-  m_Size.resize(2);
-  std::fill(m_Index.begin(), m_Index.end(), 0);
-  std::fill(m_Size.begin(), m_Size.end(), 0);
-}
 
 ImageIORegion ::~ImageIORegion() = default;
 
 ImageIORegion ::ImageIORegion(unsigned int dimension)
-{
-  m_ImageDimension = dimension;
-  m_Index.resize(m_ImageDimension);
-  m_Size.resize(m_ImageDimension);
-  std::fill(m_Index.begin(), m_Index.end(), 0);
-  std::fill(m_Size.begin(), m_Size.end(), 0);
-}
+  : m_ImageDimension{ dimension }
+  , m_Index(dimension)
+  , m_Size(dimension)
+{}
 
-ImageIORegion ::ImageIORegion(const Self & region)
-  : Region()
-{
-  m_Index = region.m_Index;
-  m_Size = region.m_Size;
-  m_ImageDimension = region.m_ImageDimension;
-}
 
-void
+ImageIORegion &
 ImageIORegion ::operator=(const Self & region)
 {
-  m_Index = region.m_Index;
-  m_Size = region.m_Size;
```

Comment:
FYI Here is where the original `ImageIORegion` assignment operator was exception unsafe: if `m_Size = region.m_Size` would throw an exception, it could leave `this` object in a corrupted state: `m_Index.size()` and `m_Size.size()` could then be different.

Reply:
Note that a compiler-generated _(defaulted)_ copy-assignment would not be exception-safe either, as it would just do a member-wise assignment. Which is why I did not follow the Rule Of Zero, this time.

Reply:
I was wondering why not follow the rule of zero this time 😄

Reply:
> I was wondering why not follow the rule of zero this time

Well... I guess there's a trade-off between code simplicity and exception-safety this time. Do we care about the scenario when the `ImageIORegion` assignment _might_ get an exception when it's only half way...? Or would the Rule Of Zero be good enough for `ImageIORegion`?

Reply:
I think Dzenan was thanking you for the explanation as these details for the motivation of the code were not self evident. Perhaps a comment is needed as to why the rule of zero was not followed.

Thank you for paying attention to these details.

Reply:
Yes, I forgot to thank for the explanation 😕